File: application/src/main/java/whenyourcar/application/service/user/UserLikeService.java
package whenyourcar.application.service.user;

import whenyourcar.domain.entity.Car;
import whenyourcar.domain.entity.User;
import org.springframework.data.domain.Pageable;
import whenyourcar.application.dto.user.like.UserLikeResponse;

public interface UserLikeService {
    public void postUserLike(Car car, User user);
    public UserLikeResponse.UserLikesResponseDto getUserLikes(User user, Pageable pageable);
    public void deleteUserLike(String email, Long userLikeId);
}

File: application/src/main/kotlin/howmuchcar/application/service/user/UserLikeService.kt
package howmuchcar.application.service.user

import howmuchcar.application.dto.user.UserLikeResponse
import howmuchcar.domain.entity.Car
import howmuchcar.domain.entity.User
import org.springframework.data.domain.Pageable

interface UserLikeService {
    fun postUserLike(user: User, car: Car)
    fun getUserLikes(user: User, pageable: Pageable): UserLikeResponse
    fun deleteUserLike(user: User, userLikeId: Long)
}